lib/benchee/benchmark/runner.ex

```diff
@@ -171,11 +169,9 @@ defmodule Benchee.Benchmark.Runner do
     {run_time, memory_usage} = measure_iteration(scenario, scenario_context)
 
     if run_time >= @minimum_execution_time do
-      if is_nil(memory_usage) do
-        {num_iterations, run_time / num_iterations, memory_usage}
-      else
-        {num_iterations, run_time / num_iterations, memory_usage / num_iterations}
-      end
+      {adjusted_run_time, adjusted_memory_usage} =
+        adjust_for_iterations(run_time, memory_usage, num_iterations)
+      {num_iterations, adjusted_run_time, adjusted_memory_usage}
```

Comment:
At first I thought the logic was completely repeated... but here it has `num_iterations` as well... I still felt like it was worth it. Not 100% sure, let me know what you think :)